activerecord/lib/active_record/model_schema.rb

```diff
@@ -331,7 +332,21 @@ def real_inheritance_column=(value) # :nodoc:
       # The list of columns names the model should ignore. Ignored columns won't have attribute
       # accessors defined, and won't be referenced in SQL queries.
       def ignored_columns
-        @ignored_columns || superclass.ignored_columns
+        return @ignored_columns if @ignored_columns
+        return superclass.ignored_columns if superclass.respond_to?(:ignored_columns)
+        return column_names - @permitted_columns if schema_loaded? && @permitted_columns.present?
```

Comment:
We should avoid having different behavior based on whether the schema is loaded or not. I don't think this is necessary at all.